minor: small cleanup

kkrolczyk opened the pull request:
cleanup of translator.py and project.py
no functionality has been changed, just removed some duplication ect

Signed-off-by: Krzysztof Królczyk <Krzysztof.Krolczyk@o2.pl>

priv-kweihmann commented on flict/flictlib/translator.py:
```diff
@@ -12,12 +12,20 @@
+try:
+    from flict.flictlib import logger as logger
+except ImportError: # allows usage also as a standalone tool
+    class MonkeyPatchLogger:
```
That deserve some explanation, as I don't quite follow the given info

kkrolczyk replied:
When running the file stand-alone, it would try to import logger and fail, provided the script could be executed from the same directory where it is located. There aren't many dependencies, actually none, except for the logger. By creating mocked logger class, the error would be circumvented, while preserving original functionality (code is imported in flict/flictlib/license.py, that seems to be the only usage of the code from that file)

hesa replied:
Is there a use case for running this file stand alone?

kkrolczyk replied:
That's a great question. I don't think so - but perhaps there is a different work-flow where someone would benefit from it? If not - perhaps this, as well as `argparse` block could be removed instead.

priv-kweihmann commented on flict/flictlib/translator.py:
```diff
@@ -61,23 +69,19 @@ def parse():
-                        type=str,
-                        dest='package_file',
-                        help='file with package definition')
+#    parser.add_argument('-p', '--package-file',
```
Why disable them? Could you kindly provide some insights. Even if they're used in the current setup it's worth making the removal a separate commit

kkrolczyk replied:
I actually got an opposite impression. Please look at the comment above. Actual usage is currently limited to importing single function from that file, however there was also argparse, as it would have been intended to be used separately, from command line. However, `parse()` was never called, in case the file was executed from command line. As i didn't know the history, and more importantly the reason or idea behind that - instead of removing them i thought it might be more beneficial to keep them commented out - after enabling proper parsing from command line, an user could be surprised reading help, that graph or packaging file could be provided to the script, while in fact they weren't used, nor the code for graph creation was there. So, long story short - i thought this would prevent confusing user, prevent misleading.

priv-kweihmann commented on flict/flictlib/translator.py:
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
+        with open(translations_file) as file_:
+            translations_object = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
+        raise RuntimeError("reading translations file failed")
```
We are using our own Exception set for all kinds of exception to encountered - please have a look at https://github.com/vinland-technology/flict/blob/e2a15bdcc59d3de72558d817a26e6f4ef5344a94/flict/flictlib/return_codes.py#L50 and turn it into a FlictError with appropriate error code

kkrolczyk replied:
indeed, i have seen this in other files. However, this would add another dependency to be resolved during importing symbol. Perhaps this file was never meant to be a stand-alone script, which could be used separately?
I could use RET_FILE_NOT_FOUND and RET_INVALID_EXPRESSSION - but i think it was purposely just for invalid license expression. Should one create a new return code, for general errors in input JSON file?

priv-kweihmann commented on flict/flictlib/translator.py:
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
+                    symbols[transl] = []
+                symbols[transl].append(key)
+        else:
+            raise RuntimeError(f"Failed parsing item: {item}")
```
Same here

priv-kweihmann commented on flict/flictlib/translator.py:
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
+        with open(jsonfile) as file_:
+            packages = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
+        raise RuntimeError("reading packages file failed")
```
and here once more

priv-kweihmann commented on flict/flictlib/translator.py:
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
+    try:
+        with open(jsonfile) as file_:
+            packages = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
```
I personally would also cover PermissionError here in case the user doesn't have read access to the file

priv-kweihmann commented on flict/flictlib/translator.py:
```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
+    return packages
+
+
+if __name__ == "__main__":
```
I see that as a change of functionality - so it should come as a separate commit with some justification

kkrolczyk replied:
I'd say this is a bit of a gray area. As explained in the first comment, the file would would just add a different way to work with the file, without modifying how it was used from within flict. However, probably it might be more important to understand if this was ever the case, or could it be useful to work with it bypasing flict? Nevertheless - it could be of course reworked to be a separate commit, if it was actually useful to to have.

priv-kweihmann replied:
@hesa could you provide some insights if this was ever meant to run standalone?

priv-kweihmann reviewed (changes requested):
See comments

hesa commented:
I would rather we remove the stand alone functionality from all files (apart from the one "real" main file).

kkrolczyk commented:
> I would rather we remove the stand alone functionality from all files (apart from the one "real" main file).

Sure, done :)

priv-kweihmann commented on flict/flictlib/translator.py:
```diff
@@ -94,61 +28,62 @@ def update_license(translations, license_expr):
+    results = {'original': translations_object,
+               'translations_map': translations_map,
+               'symbols': symbols}
+    # enabling this crashes lots of tests :)
```
What tests are we talking about?

kkrolczyk replied:
replied below

jens-erdmann commented:
This is no longer a small cleanup. The commit message doesn't reflect on the actual changes anymore. Cramming all these changes into one commit also doesn't sit well with me. Can that be split up a bit?

kkrolczyk commented:
@jens-erdmann  sure, so far i see 2 major things:
- cosmetics in project.py and relicense.py
- removal of unused parse() and duplicated code inside translator.py - probably this would be the split point


@priv-kweihmann  failing tests

```diff --git a/flict/flictlib/translator.py b/flict/flictlib/translator.py
index 99fcbc2..8d370fe 100755
--- a/flict/flictlib/translator.py
+++ b/flict/flictlib/translator.py
@@ -67,9 +67,9 @@ def read_translations(translations_file):
                'translations_map': translations_map,
                'symbols': symbols}
     # enabling this crashes lots of tests :)
-    # return results
-    read_translations.results = results
-    return symbols
+    return results
+    # read_translations.results = results
+    # return symbols
```

ERROR: tests/test_outbounds.py::test_outbound
ERROR: tests/test_verify.py::test_verify_file
ERROR: tests/test_outbounds.py::test_no_relicense
ERROR: tests/test_verify.py::test_verify_expression
ERROR: tests/test_report.py::ReportTest::test_outbound
FAIL: tests/test_project.py::Basic::test_small
FAIL: tests/test_compat.py::test_compat
FAIL: tests/test_project.py::Basic::test_big
FAIL: tests/test_simplified.py::test_simplify
FAILED (errors=6, failures=4)

tracedumps are quite big, so i'll skip posting them here. Given the current tests  

some are quite confusing:
```
Exception: BSD is not a supported license.
During handling of the above exception, another exception occurred:
...
flict.flictlib.return_codes.FlictError: (<ReturnCodes.RET_INVALID_EXPRESSSION: (11, 'Invalid expression')>, "Could not parse license expression: ['MIT BSD GPL-2.0-only WITH Classpath-exception-2.0']")
```

so basically, i think some tests assume that single "symbols" are returned from the read_translations(translations_file) function.

I've done a quick search for usage of functions from `translator.py` file:
- read_packages_file() is unused
- read_translations() is used by `license.py`
- update_{packages,license} are used only within `translator.py`
- this includes searching in tests directory

priv-kweihmann commented:
> @jens-erdmann sure, so far i see 2 major things:
> 
> * cosmetics in project.py and relicense.py
> * removal of unused parse() and duplicated code inside translator.py - probably this would be the split point
> 
> @priv-kweihmann failing tests
> 
> ```diff
> index 99fcbc2..8d370fe 100755
> --- a/flict/flictlib/translator.py
> +++ b/flict/flictlib/translator.py
> @@ -67,9 +67,9 @@ def read_translations(translations_file):
>                 'translations_map': translations_map,
>                 'symbols': symbols}
>      # enabling this crashes lots of tests :)
> -    # return results
> -    read_translations.results = results
> -    return symbols
> +    return results
> +    # read_translations.results = results
> +    # return symbols
> ```
> 
> ERROR: tests/test_outbounds.py::test_outbound ERROR: tests/test_verify.py::test_verify_file ERROR: tests/test_outbounds.py::test_no_relicense ERROR: tests/test_verify.py::test_verify_expression ERROR: tests/test_report.py::ReportTest::test_outbound FAIL: tests/test_project.py::Basic::test_small FAIL: tests/test_compat.py::test_compat FAIL: tests/test_project.py::Basic::test_big FAIL: tests/test_simplified.py::test_simplify FAILED (errors=6, failures=4)
> 
> tracedumps are quite big, so i'll skip posting them here. Given the current tests
> 
> some are quite confusing:
> 
> ```
> Exception: BSD is not a supported license.
> During handling of the above exception, another exception occurred:
> ...
> flict.flictlib.return_codes.FlictError: (<ReturnCodes.RET_INVALID_EXPRESSSION: (11, 'Invalid expression')>, "Could not parse license expression: ['MIT BSD GPL-2.0-only WITH Classpath-exception-2.0']")
> ```
> 
> so basically, i think some tests assume that single "symbols" are returned from the read_translations(translations_file) function.
> 
> I've done a quick search for usage of functions from `translator.py` file:
> 
> * read_packages_file() is unused
> * read_translations() is used by `license.py`
> * update_{packages,license} are used only within `translator.py`
> * this includes searching in tests directory

Ok - guess that looks like another refactoring incoming :disappointed: -- @hesa FYI (as I guess that one is one you)
@kkrolczyk I'm okay with solving it like it is right now (but as Jens already suggested it should come as a separated commit)

kkrolczyk committed:
minor: small cleanup

Signed-off-by: Krzysztof Królczyk <Krzysztof.Krolczyk@o2.pl>

kkrolczyk committed:
translator.py: cleanup

combined read_translations and _read_translations
added FlictError file error handling
removed unused parse function

Signed-off-by: Krzysztof Królczyk <Krzysztof.Krolczyk@o2.pl>
